File: ios/Sabagiro/AppConfig.swift
import Foundation

enum AppConfig {
  /// Production site loaded in the app shell.
  static let siteURL = URL(string: "https://www.sabagiro.ge")!

  private static let sabagiroHostSuffixes = ["sabagiro.ge", "vercel.app"]
  private static let paymentGatewayHostSuffixes = ["flitt.com"]

  #if DEBUG
  /// Simulator/local Next.js: use http://127.0.0.1:3001 or LAN IP on device.
  static let allowsLocalhost = true
  #else
  static let allowsLocalhost = false
  #endif

  static func isSabagiroHost(_ host: String) -> Bool {
    let h = host.lowercased()
    return sabagiroHostSuffixes.contains { h == $0 || h.hasSuffix(".\($0)") }
  }

  static func isLocalDevHost(_ host: String) -> Bool {
    #if DEBUG
    guard allowsLocalhost else { return false }
    let h = host.lowercased()
    return h == "localhost" || h == "127.0.0.1" || h.hasPrefix("192.168.")
    #else
    return false
    #endif
  }

  static func isPaymentGatewayHost(_ host: String) -> Bool {
    let h = host.lowercased()
    return paymentGatewayHostSuffixes.contains { h == $0 || h.hasSuffix(".\($0)") }
  }

  /// Keep navigation inside WKWebView (not Safari).
  static func shouldStayInApp(url: URL, paymentCheckoutActive: Bool) -> Bool {
    guard let scheme = url.scheme?.lowercased(), scheme == "http" || scheme == "https" else {
      return false
    }
    let host = url.host?.lowercased() ?? ""
    if isSabagiroHost(host) || isLocalDevHost(host) { return true }
    if isPaymentGatewayHost(host) { return true }
    // 3-D Secure / bank pages during Flitt checkout
    if paymentCheckoutActive { return true }
    return false
  }

  /// `true` = payment flow started, `false` = returned to site after pay.
  static func paymentFlowStateChange(for url: URL) -> Bool? {
    let host = url.host?.lowercased() ?? ""
    let path = url.path.lowercased()

    if isPaymentGatewayHost(host) {
      return true
    }

    if isSabagiroHost(host) || isLocalDevHost(host) {
      if path.hasPrefix("/payment/") {
        return true
      }
      if path == "/account" || path.hasPrefix("/account/") {
        return false
      }
      if path == "/cart" || path == "/events" || path.hasPrefix("/events/") {
        return false
      }
    }

    return nil
  }
}

File: android/app/src/main/java/ge/sabagiro/app/AppConfig.kt
package ge.sabagiro.app

import android.net.Uri

object AppConfig {
    /** Production site loaded in the app shell. */
    val siteUrl: Uri = Uri.parse("https://www.sabagiro.ge/")

    private val sabagiroHostSuffixes = listOf("sabagiro.ge", "vercel.app")
    private val paymentGatewayHostSuffixes = listOf("flitt.com")

    private val allowsLocalhost: Boolean = BuildConfig.DEBUG

    fun isSabagiroHost(host: String): Boolean {
        val h = host.lowercase()
        return sabagiroHostSuffixes.any { suffix -> h == suffix || h.endsWith(".$suffix") }
    }

    fun isLocalDevHost(host: String): Boolean {
        if (!allowsLocalhost) return false
        val h = host.lowercase()
        return h == "localhost" || h == "127.0.0.1" || h.startsWith("192.168.") || h.startsWith("10.0.2.2")
    }

    fun isPaymentGatewayHost(host: String): Boolean {
        val h = host.lowercase()
        return paymentGatewayHostSuffixes.any { suffix -> h == suffix || h.endsWith(".$suffix") }
    }

    /** Keep navigation inside the WebView (not Chrome Custom Tabs). */
    fun shouldStayInApp(uri: Uri, paymentCheckoutActive: Boolean): Boolean {
        val scheme = uri.scheme?.lowercase() ?: return false
        if (scheme != "http" && scheme != "https") return false
        val host = uri.host?.lowercase() ?: return false
        if (isSabagiroHost(host) || isLocalDevHost(host)) return true
        if (isPaymentGatewayHost(host)) return true
        if (paymentCheckoutActive) return true
        return false
    }

    /** `true` = payment flow started, `false` = returned to site after pay. */
    fun paymentFlowStateChange(uri: Uri): Boolean? {
        val host = uri.host?.lowercase() ?: return null
        val path = uri.path?.lowercase() ?: "/"

        if (isPaymentGatewayHost(host)) return true

        if (isSabagiroHost(host) || isLocalDevHost(host)) {
            if (path.startsWith("/payment/")) return true
            if (path == "/account" || path.startsWith("/account/")) return false
            if (path == "/cart" || path == "/events" || path.startsWith("/events/")) return false
        }

        return null
    }
}
